app/services/warm_provider_cache.rb

```diff
@@ -0,0 +1,39 @@
+class WarmProviderCache
+  include VendorAPI::ApplicationDataConcerns
+
+  attr_reader :version_number
+
+  def call(api_version, provider_id)
+    @provider = Provider.find(provider_id)
+    @version_number = api_version
+    r = { @provider.name => [] }
+    # For each of the application choices generate a cache entry
+    application_choices_visible_to_provider
+      .where('application_choices.updated_at between (?) and (?)', Date.new(2024, 9, 2), Date.new(2024, 9, 5))
```

Comment:
Can this all be move to a query class or scope?

Reply:
Not really, it's not reusable